### PR TITLE
Crimre 180 use review data for workload report

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

Use local data for workload report.

## Link to relevant ticket

As part of: [CRIMRE-180](https://dsdmoj.atlassian.net/browse/CRIMRE-180)

## Notes for reviewer

Adds ReceivedOnReport read model to  keep count of how many applications from a given business day are outstanding.
 
Updates the WorkloadReport to use this data rather than the datastore.

To ensure consistency with the datastore, application received events record and use the submitted_at from the datastore.

## How to manually test the feature

[CRIMRE-180]: https://dsdmoj.atlassian.net/browse/CRIMRE-180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: spec/aggregates/reviewing/review_spec.rb [hunk 1]
```diff
@@ -36,8 +38,8 @@
     let(:user_id) { SecureRandom.uuid }
 
     before do
-      review.receive_application(application_id:)
-      review.send_back(application_id:, user_id:, reason:)
+      review.receive_application(submitted_at:)
+      review.send_back(user_id:, reason:)
     end
 
     it 'is becomes "sent_back"' do
```
Comment (willmcb) [inline spec/aggregates/reviewing/review_spec.rb:45]: Not changed in this PR, but the grammar is wrong here "it is becomes "sent_back" --> it becomes "sent_back"

File: spec/aggregates/reviewing/review_spec.rb [hunk 2]
```diff
@@ -67,8 +69,8 @@
     let(:user_id) { SecureRandom.uuid }
 
     before do
-      review.receive_application(application_id:)
-      review.complete(application_id:, user_id:)
+      review.receive_application(submitted_at:)
+      review.complete(user_id:)
     end
 
     it 'is becomes "completed"' do
```
Comment (willmcb) [inline spec/aggregates/reviewing/review_spec.rb:76]: Same here

File: spec/models/reporting/processed_report_spec.rb [hunk 3]
```diff
@@ -7,19 +7,19 @@
   subject(:report) { described_class.new }
 
   describe '#table' do
-    it 'returns a table of report data' do
+    it 'returns a table report data' do
```
Comment (willmcb) [inline spec/models/reporting/processed_report_spec.rb:10]: It returns a table of report data?

File: spec/models/reporting/workload_report_spec.rb [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+      describe 'closed applications' do
+        subject(:closed_counts) { columns.last.map(&:content) }
+
+        it { is_expected.to eq([5, 8, 1, 1051]) }
+      end
+    end
+  end
+
+  describe 'number of rows can be configure using :number_of_days' do
```
Comment (willmcb) [inline spec/models/reporting/workload_report_spec.rb:64]: configured*

**Review by willmcb — APPROVED** [review]
👍 few small comments on some of the wording in the specs

**Review by arthurashman — APPROVED** [review]

**Pull request merged**